learn/core_concepts/primary_key.md

```diff
@@ -141,40 +141,40 @@ The code below updates the primary key to `title`:
 
 ### Meilisearch guesses your primary key
 
-Suppose you add documents to an index without previously setting its primary key. In this case, Meilisearch will automatically look for an attribute containing the string `id` in a case-insensitive manner (for example, `uid`, `BookId`, `ID`, `123id123`) in your first document and set it as the index's primary key.
+Suppose you add documents to an index without previously setting its primary key. In this case, Meilisearch will automatically look for an attribute ending with the string `id` in a case-insensitive manner (for example, `uid`, `BookId`, `ID`) in your first document and set it as the index's primary key.
 
-If Meilisearch cannot find a suitable attribute, the document addition process will be interrupted and no documents will be added to your index.
+If Meilisearch finds multiple attributes ending with `id`, it will throw the [`index_primary_key_multiple_candidates_found`](#index-primary-key-multiple-candidates-found) error and ask you to manually set the primary key. If it cannot find a suitable attribute, it will throw the [`index_primary_key_no_candidate_found`](#index-primary-key-no-candidate-found) error. In both cases, the document addition process will be interrupted and no documents will be added to your index.
 
 ## Primary key errors
 
 This section covers some primary key errors and how to resolve them.
 
-### `primary_key_inference_failed`
+### `index_primary_key_no_candidate_found`
 
-This error occurs when you add documents for the first time and Meilisearch [fails to guess your primary key](#meilisearch-guesses-your-primary-key). It can be resolved by [manually setting the index's primary key](#setting-the-primary-key-on-document-addition), or ensuring that all documents you add possess an `id` attribute.
+This error occurs when you add documents for the first time and none of them have an attribute ending with `id`. It can be resolved by [manually setting the index's primary key](#setting-the-primary-key-on-document-addition), or ensuring that all documents you add possess an `id` attribute.
```

Comment:
```suggestion
This error occurs when you add documents to an index for the first time and none of them have an attribute ending with `id`. It can be resolved by [manually setting the index's primary key](#setting-the-primary-key-on-document-addition), or ensuring that all documents you add possess an `id` attribute.
```